tests/admin.test.ts

```diff
@@ -0,0 +1,91 @@
+import { ActivityScope, ActivityType, UserAccessType } from '../types';
+import { ControllerFactory } from './controllers';
+import { DatabaseConnection, EventFactory, UserFactory, PortalState } from './data';
+
+beforeAll(async () => {
+  await DatabaseConnection.connect();
+});
+
+beforeEach(async () => {
+  await DatabaseConnection.clear();
+});
+
+afterAll(async () => {
+  await DatabaseConnection.clear();
+  await DatabaseConnection.close();
+});
+
+describe('retroactive attendance submission', () => {
+  test('logs activity, attendance, and points for users who have not attended', async () => {
+    const conn = await DatabaseConnection.get();
+    const [user1, user2] = UserFactory.create(2);
+    const emails = [user1, user2].map((user) => user.email);
+    const [onBehalfOfUser] = UserFactory.with({ accessType: UserAccessType.ADMIN });
```

Comment:
I know I suggested it but `onBehalfOf` is a bit misleading since the caller's doing it on behalf of other users. Apparently the legal term would be "per procuration" but that's too obscure for our purposes. Maybe let's just stick with `user` and rename `user1` and `user2` to `onBehalfOf1` and `onBehalfOf2`.

Reply:
I think the noun form of [proxy](https://www.lexico.com/definition/proxy) ("a person authorized to act on behalf of another") is a possible word, though it might be confused with the CS meaning of it. I'd rather rename just the admin instead of all the users if we can